admin/services/data-access/group.data.service.js

```diff
@@ -125,9 +130,45 @@ groupDataService.sqlUpdateGroup = async (id, name) => {
       type: TYPES.DateTimeOffset
     }
   ]
-  return sqlService.modify(`UPDATE ${sqlService.adminSchema}.[group] SET name=@name, updatedAt=@updatedAt WHERE [id]=@id`, params)
+  return sqlService.modify(
+    `UPDATE ${sqlService.adminSchema}.[group] 
+    SET name=@name, updatedAt=@updatedAt 
+    WHERE [id]=@id`,
+    params)
 }
 
 // @TODO: updating pupils for group - this now needs to be done separately.
 
+/**
+ * Get pupils filtered by school,
+ * joined by all not grouped and grouped by group_id.
+ * @param schoolId
+ * @param groupId
+ * @returns {Promise<*>}
+ */
+groupDataService.sqlGetPupils = async (schoolId, groupId) => {
+  let param = [
+    {
+      name: 'schoolId',
+      value: schoolId,
+      type: TYPES.Int
+    },
+    {
+      name: 'groupId',
+      value: groupId,
+      type: TYPES.Int
+    }
+  ]
+
+  let sql = `SELECT p.[id], p.[foreName], p.[middleNames], p.[lastName], g.[group_id]
+    FROM [mtc_admin].[pupil] p 
+    LEFT JOIN [mtc_admin].[pupilGroup] g 
+      ON p.id = g.pupil_id 
+    WHERE p.school_id=@schoolId 
+    AND g.group_id=@groupId  
+    OR g.group_id IS NULL`
```

Comment:
are you looking for _just_ the pupils in groups? or all pupils, and the group if they have one?

Reply:
Here I'm looking for all the pupils that are in the groupId passed as argument as well as the ones that are available to be in a group. This will produce a list of available pupils (we can't add a pupil to more than one group) as well as the ones already part of the current group (so that the checkbox is checked).
The code has changed a bit though, it's coming in the next push.